tests/js/spec/components/smartSearchBar/index.spec.jsx

```diff
@@ -984,6 +985,108 @@ describe('SmartSearchBar', function () {
     });
   });
 
+  describe('getTagKeys()', function () {
+    const supportedTagsMock = {
+      has: {
+        key: 'has',
+      },
+      aa: {
+        key: 'aa',
+      },
+      bb: {
+        key: 'bb',
+      },
+    };
+
+    let spy;
+    beforeAll(() => {
+      spy = jest.spyOn(Fields, 'getFieldDefinition').mockImplementation(key => {
```

Comment:
@jennmueng good test, but I think I would rather not mock out the fields here. We should avoid mocking as much as possible so we can catch anything that goes wrong in the fields file. The downside is that we might break this test by modifying field definitions, but I think that's better than missing a potential bug.

Reply:
Ah I see, originally went through with mocking because I thought we wanted to specifically test this function rather than the field descriptions themselves.

Reply:
You're not necessarily wrong going for a unit test, but our component tests are a mix of unit and integration tests (leaning more to integration). I generally adhere to the KCD mantra that integration tests have the best return on investment: https://kentcdodds.com/blog/write-tests

Reply:
I see, thanks for the insight!